core/src/labels/label.cpp

```diff
@@ -30,38 +32,80 @@ Label::Label(Label::Transform _transform, glm::vec2 _size, Type _type, Options _
 
 Label::~Label() {}
 
-bool Label::updateScreenTransform(const glm::mat4& _mvp, const glm::vec2& _screenSize, bool _drawAllLabels) {
-
-    glm::vec2 screenPosition;
+bool Label::updateScreenTransform(const glm::mat4& _mvp, float _tileInverseScale,
+                                  const ViewState& _viewState, bool _drawAllLabels) {
     bool clipped = false;
 
     switch (m_type) {
         case Type::debug:
         case Type::point:
         {
-            glm::vec2 p0 = m_transform.modelPosition1;
+            glm::vec2 p0 = m_worldTransform.position;
 
-            screenPosition = worldToScreenSpace(_mvp, glm::vec4(p0, 0.0, 1.0),
-                                                _screenSize, clipped);
-            if (clipped) {
-                return false;
-            }
+            if (m_options.flat) {
+                glm::vec2 positions[4];
+                float unitsPerMeter = _viewState.metersPerPixel * _tileInverseScale;
+                glm::vec2 metersDimension = m_dim * unitsPerMeter;
+
```

Comment:
Why calling this metersDimension? I think you calculate `unitsPerPixel` here. Keeping this all pixel based would simplify things (for me).